testing/test_recwarn.py

```diff
@@ -238,6 +238,16 @@ def test_record_only(self):
         assert str(record[0].message) == "user"
         assert str(record[1].message) == "runtime"
 
+    def test_record_by_subclass(self):
```

Comment:
Could you please also add a test which checks for warning subclasses when the user passes more than one warning to `pytest.warns`? For example:

```python
class Warn1(UserWarning): pass
class Warn2(RuntimeWarning): pass

with pytest.warns((UserWarning, RuntimeWarning)):
   warnings.warn("w1", Warn1)
   warnings.warn("w2", Warn2)

assert len(record) == 2
...
```